rectools/models/sasrec.py

```diff
@@ -561,13 +540,57 @@ def get_dataloader_recommend(self, dataset: Dataset) -> DataLoader:
         return recommend_dataloader
 
 
+# ####  --------------  Lightning Model  --------------  #### #
+
+
+class SessionEncoderLightningModule(SessionEncoderLightningModuleBase):
+    """TODO"""
+
+    def on_train_start(self) -> None:
+        """TODO"""
+        self._xavier_normal_init()
+
+    def training_step(self, batch: torch.Tensor, batch_idx: int) -> torch.Tensor:
+        """TODO"""
+        x, y, w = batch
+        logits = self.forward(x)  # [batch_size, session_maxlen, n_items + 1]
+        # We are using CrossEntropyLoss with a multi-dimensional case
```

Comment:
These comments apply only to softmax loss so we need to place them after `if`

Reply:
fixed